stan/math/prim/scal/fun/is_nan.hpp

```diff
@@ -14,7 +15,11 @@ namespace math {
  * @param x Value to test.
  * @return <code>true</code> if the value is NaN.
  */
-inline bool is_nan(double x) { return std::isnan(x); }
+template <typename T>
+inline typename std::enable_if_t<std::is_arithmetic<T>::value, bool> is_nan(
```

Comment:
Same comment here as in the issue for changing all of these.  At most, this should be

```
template <typename T>
inline typename std::enable_if_t<std::is_arithmetic<T>::value, bool> is_nan(T x) {
  return std::isnan(x);
}
```

so primitives get passed by value.